scripts/colab_helper.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+###############################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+###############################################################################
+"""
+Install IDAES, Ipopt, and other solvers on Google Colab
+
+Created by Alex Dowling (adowling@nd.edu) and Jeff Kantor at the University of Notre Dame
+with input from John Siirola at Sandia National Laboratories.
+
+To use this script, add the following to a code block in a Jupyter notebook:
+
+```
+# Ipopt installer
+import sys
+# If running on Google Colab, install Pyomo and Ipopt via IDAES
+if "google.colab" in sys.modules:
+    !wget "https://raw.githubusercontent.com/idaes-pse/main/scripts/colab_helper.py"
+    import colab_helper
+    colab_helper.install_idaes()
+    colab_helper.install_ipopt()
+```
+
+For testing purposes, you may need to use:
+"https://raw.githubusercontent.com/adowling2/idaes-pse/colab-install-script/scripts/colab_helper.py"
+
+"""
+
+import shutil
+import sys
+import os.path
+import os
+import re
+
+import subprocess
+
+
+def _check_available(executable_name):
+    """Utility to check in an executable is available"""
+    return shutil.which(executable_name) or os.path.isfile(executable_name)
+
+
+def package_available(package_name):
+    """Utility to check if a package/executable is available
+
+    This supports customization, e.g., glpk, for special package names
+    """
+
+    if package_name == "glpk":
+        return _check_available("gpsol")
+    else:
+        return _check_available(package_name)
+
+
+def on_colab():
+    """Utility returns True if executed on Colab, False otherwise"""
+    return "google.colab" in sys.modules
+
+
+def install_idaes():
+    """Installs latest version of IDAES-PSE via pip"""
+
+    try:
+        import idaes
+
+        print("idaes was found! No need to install.")
+    except ImportError:
+        print("Installing idaes via pip...")
+        subprocess.run(
+            [sys.executable, "-m", "pip", "install", "-q", "idaes_pse"], check=True
+        )
+        print("idaes was successfully installed")
+        v = subprocess.run(
+            ["idaes", "--version"], check=True, capture_output=True, text=True
+        )
+        print(v.stdout)
+        print(v.stderr)
```

Comment:
If you are just going to print out the output anyway, why bother with `capture=True`?

Reply:
On the terminal, the output comes to the terminal if you don't specify `capture_output`.  I wonder if this is different for the collab environment?  If that's the case, we want to at least be consistent and capture/print for *all* subprocess calls.

Reply:
I expected that behavior on Colab, but no luck when testing in a cell.

Reply:
@jsiirola, I like your suggestion. I added a verbose flag to display the captured pip, idaes get-extensions, etc. output